### DISCUSSION TITLE
Suurballe and Bhandari Algorithms

### DISCUSSION THREAD

**Discussion by HussamAwad12** [networkx/networkx#7722] [Algorithms]
Hello,

I'm working on Suurballe and Bhandari algorithms as part of my university project, one of the project aims is to contribute to NetworkX library
suurballe finds 2 edge-disjoint minimum sum paths between two nodes, and bhandari is more general for k paths
I'd like to contribute the code to networkx (which is still in development), who can I speak with to discuss the idea, and if the idea is approved, code standards for the library?

**Comment by HussamAwad12** [comment]
also, if you have NetworkX algorithms or other code that needs to be implemented, I'd happy to do as part of my project

**Comment by dschult** [comment]
A routine to compute two edge-disjoint shortest paths (and k paths) sounds like a reasonable function for networkx. The algorithms you mention also seem like the natural ones to implement.  We don't require prior approval for you to submit the work for review. I would suggest putting it in a new module (maybe `multiple_paths.py`) in the folder `networkx/algorithms/shortest_paths/` directory with tests in a module (`test_multiple_paths.py`?) in the tests directory of that same folder. Looks at other modules in those folders to get a general idea of how they are used. You will need to know enough `git` to effectively submit your code. And you should read the docs for contributing (the [CONTRIBUTING.rst](https://networkx.org/documentation/stable/developer/contribute.html) and the more general [developer guideline pages](https://networkx.org/documentation/stable/developer/index.html) ). If you have questions, you can post them here, or open a partially completed PR where a discussion can occur alongside the code you  are working on.
